pkg/importer/vddk-datasource.go

```diff
@@ -248,18 +248,24 @@ func createVMwareClient(endpoint string, accessKey string, secKey string, thumbp
 	}
 
 	// Construct VMware SDK URL and get MOref
-	sdkURL := vmwURL.Scheme + "://" + accessKey + ":" + secKey + "@" + vmwURL.Host + "/sdk"
+	sdkURL := vmwURL.Scheme + "://" + url.PathEscape(accessKey) + ":" + url.PathEscape(secKey) + "@" + vmwURL.Host + "/sdk"
+	redacted := vmwURL.Scheme + "://" + url.PathEscape(accessKey) + ":" + "*****" + "@" + vmwURL.Host + "/sdk"
 	vmwURL, err = url.Parse(sdkURL)
 	if err != nil {
-		klog.Errorf("Unable to create VMware URL: %v", err)
-		return nil, err
+		return nil, errors.New("unable to create VMware URL from: " + redacted)
 	}
```

Comment:
I don't think that `PathEscape()` is doing what we need here. For example it does not encode `@`. I believe that the right thing to do is to construct a `url.URL` and use its String() function. Encoding is a tricky business that should not be done ad hoc.

I did not try it, but I think that instead of rebuilding a string and reparsing it, we should just modify the field that we wish to.
```suggestion
         vmwURL.User = url.UserPassword(accessKey, secKey)
         vmwURL.Path = "sdk"
```

Reply:
Thank you! This is much better.